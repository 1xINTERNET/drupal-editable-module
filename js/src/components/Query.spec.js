/* eslint-env jest */

import React from "react";
import { shallow } from "enzyme";
import * as reduxJsonApi from "redux-json-api";

import { QueryPresentational } from ".";

const readEndpointSpy = jest
  .spyOn(reduxJsonApi, "readEndpoint")
  .mockImplementation(k => k);

const dispatchFn = jest.fn(() => ({
  body: {
    data: "entityData"
  }
}));

const BUNDLE_PROP = "bundle";
const TYPE_PROP = "type";
const UUID_PROP = "uuid";

describe("Query", () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <QueryPresentational
        bundle={BUNDLE_PROP}
        type={TYPE_PROP}
        dispatch={dispatchFn}
        uuid={UUID_PROP}
      >
        {() => null}
      </QueryPresentational>
    );
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call readEndpoint with the correct endpoint parameter when fetching", () => {
    expect(readEndpointSpy).toHaveBeenCalledWith(
      `${BUNDLE_PROP}/${TYPE_PROP}/${UUID_PROP}`
    );
  });
});
