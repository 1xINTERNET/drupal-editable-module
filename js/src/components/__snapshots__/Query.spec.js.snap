// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query should render correctly 1`] = `
<Connect(DataSetPresentational)
  loading={false}
  refetch={[Function]}
  resourceIds="entityData"
>
  <Component />
</Connect(DataSetPresentational)>
`;
